src/utils/validators/addWallet.ts

```diff
@@ -58,6 +58,29 @@ export const isNameUnique = server$(async function (name: string) {
   return true;
 });
 
+export const UniqueAddressResult = z.array(z.number());
+export const isAddressUnique = server$(async function (address: string) {
+  const db = await connectToDB(this.env);
+  const cookie = this.cookie.get("accessToken");
+  if (!cookie) {
+    throw new Error("No cookie found");
+  }
+  const { userId } = jwt.decode(cookie.value) as JwtPayload;
+  const result = (
+    await db.query(`
+     LET $walletId = SELECT VALUE ->observes_wallet.out FROM ${userId};
+     LET $wallets = SELECT address FROM wallet WHERE $walletId;
+     SELECT VALUE count() FROM $wallets WHERE address = type::string('${address}');
```

Comment:
Could we aspire to shorten the query to look more similiar to:
```
const balanceToUpdate = (
    await db.query(
      `SELECT * FROM balance WHERE ->(for_wallet WHERE out.address = '${getAddress(accountAddress)}') AND ->(for_token WHERE out.address = '${getAddress(tokenAddress)}');`,
    )
  ).at(0);
  ```

Reply:
putting it here for the record.
Following your proposal the query would look as follows:
```
SELECT VALUE count() FROM wallet WHERE (
    SELECT ->observes_wallet.out FROM "${user}"
) AND address = type::string('${address}');
```

I run a quick benchmark and it seems that despite having longer query, using LET's improves code performance.
Current query: 0.3us on average
Modified one (one above):  1.1ms on average

Reply:
Great research!
We will stay with your approach in that case.